analyzers/tests/SonarAnalyzer.UnitTest/Rules/TypeNamesShouldNotMatchNamespacesTest.cs

```diff
@@ -27,18 +27,20 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class TypeNamesShouldNotMatchNamespacesTest
     {
+        private readonly VerifierBuilder builder = new VerifierBuilder<TypeNamesShouldNotMatchNamespaces>();
+
         [TestMethod]
         public void TypeNamesShouldNotMatchNamespaces() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\TypeNamesShouldNotMatchNamespaces.cs", new TypeNamesShouldNotMatchNamespaces());
+            builder.AddPaths("TypeNamesShouldNotMatchNamespaces.cs").Verify();
 
 #if NET
         [TestMethod]
         public void TypeNamesShouldNotMatchNamespaces_CSharp9() =>
-            OldVerifier.VerifyAnalyzerFromCSharp9Library(@"TestCases\TypeNamesShouldNotMatchNamespaces.CSharp9.cs", new TypeNamesShouldNotMatchNamespaces());
+            builder.AddPaths("TypeNamesShouldNotMatchNamespaces.CSharp9.cs").WithOptions(ParseOptionsHelper.FromCSharp9).Verify();
 
         [TestMethod]
         public void TypeNamesShouldNotMatchNamespaces_CSharp10() =>
-            OldVerifier.VerifyAnalyzerFromCSharp10Library(@"TestCases\TypeNamesShouldNotMatchNamespaces.CSharp10.cs", new TypeNamesShouldNotMatchNamespaces());
+            builder.AddPaths("TypeNamesShouldNotMatchNamespaces.CSharp10.cs").WithOptions(ParseOptionsHelper.FromCSharp10).WithConcurrentAnalysis(false).Verify();
```

Comment:
We can get rid of this limitation if we also remove the useless `namespace Tests.Diagnostics;` from the test case
```suggestion
            builder.AddPaths("TypeNamesShouldNotMatchNamespaces.CSharp10.cs").WithOptions(ParseOptionsHelper.FromCSharp10).Verify();
```

Reply:
I don't know what's best here. The AST with a NamespaceDeclaration and an AST with a FileScopedNamespaceDeclaration are almost identical. So there is not much that can go wrong. On the other hand, FileScopedNamespaceDeclaration is a new SyntaxKind, and therefore, it should be included in the test.
I think the other tests already cover concurrency, so I would vote for keeping the FileScopedNamespace.

Reply:
It doesn't make sense to use every language feature in every rule. We've already done specifications of what features affect what rules. File scoped namespaces will be addressed in [MMF-2558](https://jira.sonarsource.com/browse/MMF-2558).

Let's remove it here, it adds pain.

Reply:
This one isn't addressed yet.